### ISSUE TITLE
Wire protocol support in behave

### ISSUE THREAD

**Issue by sieben** [behave/behave#508]
I would like to know if there is a plan to support the cucumber wire protocol in behave?

https://github.com/cucumber/cucumber/wiki/Wire-Protocol

**Comment by jenisys** [comment]
@sieben 
It is on the list of features that would be nice to have.
Currently, I have no time to support it in the near future.
If you are interested, you can provide a pull-request.

Do you have a real use case?

**Comment by sieben** [comment]
I am considering using BDD methodology to test interoperability and conformance testing of network protocols. In some cases, the steps will involve remote procedures on distant hosts and it might be nice to have a way to trigger remotely some code.

**Comment by jenisys** [comment]
Mmh, that sounds more like you want to (mis)use the wire protocol as remoting protocol instead of using a real remoting protocol (or ssh login, etc.). The wire protocol just decouples feature execution from the local host to another host (and/or platform) where the steps are executed. This is normally a one-to-one coupling (AFAIK). It is basically like a parallel runner but with only one backend process/thread (on another host). From your description, you may need a more flexible or more complicated relationships either within a Scenario or Feature.

**Comment by sieben** [comment]
Maybe, that said, I'm trying to have a complete view of what BDD and behave
can provide. Feature execution and test execution could happen in my case
in different locations depending on the test I want to execute. Also,
several tests could also happen in parallel and use of multiprocessing
functionality could be useful.
